lib/std/heap.zig

```diff
@@ -8,6 +8,129 @@ const c = std.c;
 const Allocator = std.mem.Allocator;
 const windows = std.os.windows;
 
+const page_size_darwin: ?comptime_int = if (builtin.os.tag.isDarwin())
+    switch (builtin.cpu.arch) {
+        .x86, .x86_64 => 4 << 10,
+        .thumb, .thumbeb, .arm, .armeb, .aarch64, .aarch64_be, .aarch64_32 => 16 << 10,
+        else => null,
+    }
+else
+    null;
+
+// -- <https://devblogs.microsoft.com/oldnewthing/20210510-00/?p=105200>
+const page_size_windows: ?comptime_int = if (builtin.os.tag == .windows and builtin.os.version_range.windows.min.isAtLeast(.xp))
+blk: {
+    break :blk switch (builtin.cpu.arch) {
+        .x86, .x86_64 => 4 << 10,
+        // SuperH => 4 << 10,
+        .mips, .mipsel, .mips64, .mips64el => 4 << 10,
+        .powerpc, .powerpcle, .powerpc64, .powerpc64le => 4 << 10,
+        // DEC Alpha => 8 << 10,
+        // Itanium => 8 << 10,
+        .thumb, .thumbeb, .arm, .armeb, .aarch64, .aarch64_be, .aarch64_32 => 4 << 10,
+        else => null,
+    };
+} else null;
+
+const page_size_os: ?comptime_int = page_size_darwin orelse page_size_windows;
+
+/// A compile time known upper bound on page size.
```

Comment:
Just to be clear: Are we saying that if the page size is greater than this at run time, the standard library will not work? Or are we saying that this is the highest page size we're *currently* aware of being possible on the target platform, but that's all?

The doc comment should be explicit about this.

Reply:
It is both, a limit that would trip assertions if exceeded, and also the maximum value it is based on. I'm not sure what distinction to make. How about this?
```
/// The largest page size the architecture/OS might be using at runtime.
```

Reply:
Something like: "The largest page size that the target architecture + operating system might use at run time. The standard library will not work if the operating system unexpectedly exceeds this value." The second part is the important bit that needs to be clearly documented.

Reply:
> The standard library will not work if the operating system unexpectedly exceeds this value." The second part is the important bit that needs to be clearly documented.

Is this a necessary or useful restriction? What depends on this value?